src/main/java/org/jabref/logic/importer/fileformat/PdfMergeMetadataImporter.java

```diff
@@ -161,17 +161,67 @@ private void fetchData(BibEntry candidate, StandardField field, IdBasedFetcher f
     }
 
     private static BibEntry mergeCandidates(Stream<BibEntry> candidates) {
-        final BibEntry entry = new BibEntry();
-        candidates.forEach(entry::mergeWith);
+        // Convert the stream to a list so we can iterate over the list twice
```

Comment:
The "why" is missing. 

May be better: Move the "mergeWith" to your loop. Comment that you need to write to a variable and therefore streams cannot be used easily